bolides/bdf.py

```diff
@@ -0,0 +1,1032 @@
+            # users can make multiple plots without constantly recreating it
+            if hasattr(self, '_showers'):
+                sdf = self._showers
+            else:
+                from . import ShowerDataFrame
+                sdf = ShowerDataFrame()
+                self._showers = sdf
+            date_info = sdf.get_dates(showers, range(min(bdf.datetime).year-1, max(bdf.datetime).year+1))
+            shower_names = date_info['shower name']
+            shower_dates = date_info.datetime
+
+        # with the given matplotlib style,
+        with plt.style.context(style):
+
+            # initialize figure and set major, minor ticks and formats
+            fig, ax = plt.subplots(figsize=figsize, dpi=300)
+            ax.xaxis.set_major_locator(mdates.YearLocator())
+            ax.xaxis.set_minor_locator(mdates.MonthLocator())
+            ax.xaxis.set_major_formatter(mdates.DateFormatter('%Y'))
+            ax.xaxis.set_minor_formatter(mdates.DateFormatter('%m'))
+
+            # make minor ticks (months) small, pad major ticks (years)
+            plt.tick_params(axis='x', which='minor', labelsize=5, pad=0, labeltop=True)
+            plt.tick_params(axis='x', which='major', pad=3, labeltop=True)
+
+            # set x-label, adding year data if data is only within one year
+            # (and hence no year data on x-axis)
+            ax.set_xlabel("Date")
+            if min(bdf.datetime).year == max(bdf.datetime).year:
+                ax.set_xlabel("Date (month in "+str(min(bdf.datetime).year)+")")
+            ax.set_ylabel("# Events")
+
+            plt.hist(datetimes, bins=index, **kwargs)
+
+            # set x-limits to data limits
+            x_min = datetime.fromisoformat(start) if start is not None else min(bdf.datetime)
+            x_max = datetime.fromisoformat(end) if end is not None else max(bdf.datetime)
+            plt.xlim(x_min, x_max)
+
+            if showers is not None:
+                names = list(shower_names)
+                unique = list(shower_names.unique())
+                import matplotlib.cm as cm
+                cmap = cm.Set2
+
+                defaults = {'linestyle': 'dashed', 'linewidth': 1,
+                            'alpha': 0.5}
+                line_kwargs = reconcile_input(line_style, defaults)
+
+                for name, date in zip(shower_names, shower_dates):
+                    plt.axvline(x=date, color=cmap(unique.index(name)/8),
+                                label=name, **line_kwargs)
+                # remove extra labels for repeat showers
+                for i, p in enumerate(ax.get_lines()):
+                    if p.get_label() in names[:i]:
+                        p.set_label('_' + p.get_label())
+                plt.legend(loc='upper right')
+
+            # make y-axis logscale if specified
+            if logscale:
+                ax.set_yscale('log')
+
+        return fig, ax
+
+    def add_website_data(self, ids=None):
+        # import json
```

Comment:
Add a header comment for `add_website_data`. 

Also, it can take a long time to downlaod the data. Can we add a status bar ad it downloads, or at least a loading... statement?

Reply:
Added a tqdm loading bar and a header comment in 7f49f433a12f5e91f535ef1069465bc24c20f665.